test/extensions/filters/http/common/compressor/compressor_integration_tests.cc

```diff
@@ -0,0 +1,296 @@
+
+  auto request_uncompressed_counter =
+      test_server_->counter("http.config_test.compressor.test.gzip.request.not_compressed");
+  ASSERT_NE(request_uncompressed_counter, nullptr);
+  ASSERT_EQ(1, request_uncompressed_counter->value());
+
+  auto response_compressed_counter =
+      test_server_->counter("http.config_test.compressor.test.gzip.compressed");
+  ASSERT_NE(response_compressed_counter, nullptr);
+  ASSERT_EQ(0, response_compressed_counter->value());
+
+  auto response_uncompressed_counter =
+      test_server_->counter("http.config_test.compressor.test.gzip.not_compressed");
+  ASSERT_NE(response_uncompressed_counter, nullptr);
+  ASSERT_EQ(1, response_uncompressed_counter->value());
+}
+
+INSTANTIATE_TEST_SUITE_P(Protocols, CompressorProxyingConnectIntegrationTest,
+                         testing::ValuesIn(HttpProtocolIntegrationTest::getProtocolTestParams()),
+                         HttpProtocolIntegrationTest::protocolTestParamsToString);
+
+void CompressorProxyingConnectIntegrationTest::initialize() {
+  config_helper_.addConfigModifier(
+      [&](envoy::extensions::filters::network::http_connection_manager::v3::HttpConnectionManager&
+              hcm) -> void { ConfigHelper::setConnectConfig(hcm, false); });
+  config_helper_.addFilter(compressorFilterConfig);
+  HttpProtocolIntegrationTest::initialize();
+}
+
+TEST_P(CompressorProxyingConnectIntegrationTest, ProxyConnect) {
+  initialize();
+
+  // Send request headers.
+  codec_client_ = makeHttpConnection(lookupPort("http"));
+  auto encoder_decoder = codec_client_->startRequest(connect_headers_);
+  request_encoder_ = &encoder_decoder.first;
+  response_ = std::move(encoder_decoder.second);
+
+  // Wait for them to arrive upstream.
+  AssertionResult result =
+      fake_upstreams_[0]->waitForHttpConnection(*dispatcher_, fake_upstream_connection_);
+  RELEASE_ASSERT(result, result.message());
+  result = fake_upstream_connection_->waitForNewStream(*dispatcher_, upstream_request_);
+  RELEASE_ASSERT(result, result.message());
+  ASSERT_TRUE(upstream_request_->waitForHeadersComplete());
+  EXPECT_EQ(upstream_request_->headers().get(Http::Headers::get().Method)[0]->value(), "CONNECT");
+  if (upstreamProtocol() == FakeHttpConnection::Type::HTTP1) {
+    EXPECT_TRUE(upstream_request_->headers().get(Http::Headers::get().Protocol).empty());
+  } else {
+    EXPECT_EQ(upstream_request_->headers().get(Http::Headers::get().Protocol)[0]->value(),
+              "bytestream");
+  }
+
+  // Send response headers
+  upstream_request_->encodeHeaders(default_response_headers_, false);
+
+  // Wait for them to arrive downstream.
+  response_->waitForHeaders();
+  EXPECT_EQ("200", response_->headers().getStatusValue());
+
+  // Make sure that even once the response has started, that data can continue to go upstream.
+  codec_client_->sendData(*request_encoder_, "hello", false);
+  ASSERT_TRUE(upstream_request_->waitForData(*dispatcher_, 5));
```

Comment:
Unlikely as it is that the compressed data would be the same size can we check for payload just in case?

ASSERT_TRUE(upstream_request_->waitForData(*dispatcher_, "hello"));